databuilder/models/table_metadata.py

```diff
@@ -59,18 +59,86 @@ def create_next_relation(self):
             return None
 
 
+class DescriptionMetadata:
+    DESCRIPTION_NODE_LABEL = 'Description'
+    PROGRAMMATIC_DESCRIPTION_NODE_LABEL = 'Programmatic_Description'
+    DESCRIPTION_KEY_FORMAT = '{description}'
+    DESCRIPTION_TEXT = 'description'
+    DESCRIPTION_SOURCE = 'description_source'
+
+    DESCRIPTION_RELATION_TYPE = 'DESCRIPTION'
+    INVERSE_DESCRIPTION_RELATION_TYPE = 'DESCRIPTION_OF'
+
+    # The default editable source.
+    DEFAULT_SOURCE = "description"
+
+    def __init__(self,
+                 source,  # type: str
+                 text  # type: Union[None, str]
+                 ):
+        """
+        :param source: The unique source of what is populating this description
+        :param text: the description text. Markdown supported.
+        """
+        self._source = source
+        self._text = text
+        # There are so many dependencies on Description node, that it is probably easier to just separate the rest out.
+        if (self._source == self.DEFAULT_SOURCE):
+            self._label = self.DESCRIPTION_NODE_LABEL
+        else:
+            self._label = self.PROGRAMMATIC_DESCRIPTION_NODE_LABEL
+
+    @staticmethod
+    def create_description_metadata(source, text):
+        # We do not want to create a node if there is no description text!
+        if text is None:
+            return None
+        if source is None or source == '':
+            description_node = DescriptionMetadata(DescriptionMetadata.DEFAULT_SOURCE, text)
+        else:
+            description_node = DescriptionMetadata(source, text)
+        return description_node
+
+    def get_description_id(self):
+        # type: () -> str
+        if self._source == self.DEFAULT_SOURCE:
+            return "_description"
+        else:
+            return "_description/" + self._source
```

Comment:
Nit: `_description/<source>` seems representing <source> that belongs to description. How about `_<source>_description` ?

Reply:
ah good point, changing.